cmd/test.go

```diff
@@ -285,8 +285,8 @@ func processWatcherUpdate(ctx context.Context, testParams testCommandParams, pat
 		})
 
 	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
-		os.Exit(1)
+		fmt.Fprintln(testParams.output, err)
+		return
```

Comment:
Solves case `2` by not terminating the watcher if it fails to read a file in the process of being written to.

I'm not entirely sure if this has unwanted effects for e.g. the `Store` state, and such, though. I'd appreciate some extra analysis here @ashutosh-narkar.

Reply:
I think we should be fine. If there is an error here the transaction will be aborted.